qiskit/transpiler/target.py

```diff
@@ -590,6 +590,245 @@ def operation_names_for_qargs(self, qargs):
+                specified with any :class:`~.Parameter` objects in the list, that entry will be
+                treated as supporting any value, however parameter names will not be checked (for
+                example if an operation in the target is listed as parameterized with ``"theta"``
+                and ``"phi"`` is passed into this function that will return properties for that
+                instruction). For example, if called with::
+
+                    parameters = [Parameter("theta")]
+                    target.get_instruction_properties_by_class(RXGate, (0,), parameters=parameters)
+
+                will return the properties of any :class:`~.RXGate` instruction that is suported on
+                qubit 0 which will accept any parameter (excluding fixed angle variants).
+
+        Returns:
+            dict: A dictionary mapping all the operation names to instruction properties for all
+                matching instructions. If there are no properties available the value will be ``None``
+
+        Raises:
+            KeyError: If the instruction isn't supported
+        """
+
+        def check_obj_params(parameters, obj):
+            for index, param in enumerate(parameters):
+                if isinstance(param, Parameter) and not isinstance(obj.params[index], Parameter):
+                    return False
+                if param != obj.params[index] and not isinstance(obj.params[index], Parameter):
+                    return False
+            return True
+
+        # Case a list if passed in by mistake
+        qargs = tuple(qargs)
+        out_dict = {}
+        for op_name, obj in self._gate_name_map.items():
+            if inspect.isclass(obj):
+                if obj != operation_class:
+                    continue
+                # If qargs set then validate no duplicates and all indices are valid on device
+                if all(qarg <= self.num_qubits for qarg in qargs) and len(set(qargs)) == len(qargs):
+                    out_dict[op_name] = None
+            elif isinstance(obj, operation_class):
+                if parameters is not None:
+                    if len(parameters) != len(obj.params):
+                        continue
+                    if not check_obj_params(parameters, obj):
+                        continue
+                if qargs in self._gate_map[op_name]:
+                    out_dict[op_name] = self._gate_map[op_name][qargs]
+                if None in self._gate_map[op_name]:
+                    out_dict[op_name] = self._gate_map[op_name][None]
+                if self._gate_map[op_name] is None:
+                    out_dict[op_name] = None
+        if not out_dict:
+            raise KeyError(
+                f"Operation class {operation_class} on qargs: {qargs} with parameters "
+                f"{parameters} is not supported"
+            )
+        return out_dict
+
+    def get_instruction_errors_by_class(self, operation_class, qargs, parameters=None):
```

Comment:
personally I would be ok not having the special methods to get `error` and `duration`.. Since error and duration can be easily accessed from `properties`, I don't think they need dedicated lookups (and they are just two properties, there could be more). This would reduce the extra added methods from 6 to 2.

Reply:
I think the purpose of this PR is to provide wrapper of these attributes. Note that `InstructionProperties` is not exposed to developers who don't update gate properties by themselves. So hiding its implementation with these methods makes sense to me.

In transpiler passes requiring these information, the pattern of `getattr(inst_props, "error", 0.0) or 0.0` is often used, so it seems to me like developers assume the case the dict value is not `InstructionProperties` (personally, I think they should use `.error` instead).

Anyways, there are several cases you cannot get error 
- instruction is not defined (1)
- qargs is not defined for the instruction (2)
- dict value is missing for (instruction, qargs): No instruction set provided (3)
- dict value is provided but not `InstructionProperties` instance: Edge case (4)
- dict value is provided but error is None value: Calibration is provided without benchmark, e.g. during gate bring-up (5)

This method returns None in all above cases. So this method reduces the complexity to access gate fidelity.

In accordance with convention, when None is returned, transpiler passes treat the instruction as ideal gate (i.e. zero-error) in the fidelity metric. I'm not sure if this is correct behavior in the case of (5). In this case, we could calculate coherence limit with device T1 and gate `.duration`, because we should know gate duration after calibration. This method can implement such logic since Target knows qubit T1.